lib/config/flat-config-schema.js

```diff
@@ -98,16 +96,21 @@ function deepMerge(first, second = {}, mergeMap = new Map()) {
     }
 
     /*
-     * First create a result object where properties from the second value
+     * First create a result object where properties from the second object
      * overwrite properties from the first. This sets up a baseline to use
      * later rather than needing to inspect and change every property
      * individually.
      */
-    const result = {
+    let result = {
         ...first,
         ...second
     };
 
+    delete result.__proto__; // eslint-disable-line no-proto -- don't merge own property "__proto__"
+    if (Array.isArray(second)) {
+        result = Object.assign([], result);
+    }
```

Comment:
Looks like this can never happen, because `second` is always a non-array object?

Reply:
Thanks! Removed in 6e2da74bf7c86b04347a742c8e702e87b9ee2644.